apps/confessions-client/components/Login.tsx

```diff
@@ -0,0 +1,88 @@
+import {
```

Comment:
https://user-images.githubusercontent.com/2636237/229979831-93f6131f-8239-4e0f-9e47-5e37ec08d014.mov

This is really close!

Here are a few things I want changed before we release this though:
- What happens if the user closes the popup window without 'proving' anything? The 'login' button remains disabled. I wonder if 1) some way to re-enable the button when the popup is closed, or 2) if we should disable that button for only a brief moment
- Logging out is instant, as it should be, nice!
- The 'Login'/'Logout' button flickers from 'Login' to 'Logout' if the user is logged in and refreshes the page. I would like it to not flicker, and say 'Logout' immediately if the user is logged in.
- There should be some intermediate feedback state between when the popup delivers a proof to the page, and when the proof is verified. For example, the button could say neither 'Login' nor 'Logout', but 'Verifying'.
- Did you test what happens if the proof that is passed back to the login flow is invalid? That should be handled gracefully as well.

Reply:
I think having an overlay saying "verifying" rather than disable the "login" button, would solve both the following problems.

> What happens if the user closes the popup window without 'proving' anything? The 'login' button remains disabled. I wonder if 1) some way to re-enable the button when the popup is closed, or 2) if we should disable that button for only a brief moment

> There should be some intermediate feedback state between when the popup delivers a proof to the page, and when the proof is verified. For example, the button could say neither 'Login' nor 'Logout', but 'Verifying'.